bin/s2nc.c

```diff
@@ -510,13 +554,27 @@ int main(int argc, char *const *argv)
             GUARD_EXIT(s2n_connection_set_session(conn, session_state, session_state_length), "Error setting session state in connection");
         }
 
+        for (size_t i = 0; i < psk_identity_count; i++) {
+            s2n_psk_hmac input_psk_hmac = atoi(psk_hmac_list[i]);
+            DEFER_CLEANUP(struct s2n_psk *input_psk = s2n_external_psk_new(), s2n_psk_free);
+
+            printf("Appending PSK Identity: %s\n", psk_identity_list[i]);
+            GUARD_EXIT(s2n_psk_set_identity(input_psk, (const uint8_t *)psk_identity_list[i], strlen(psk_identity_list[i])), "Error setting psk identity");
+            GUARD_EXIT(s2n_psk_set_secret(input_psk,  (const uint8_t *)psk_secret_list[i], strlen(psk_secret_list[i])), "Error setting psk secret");
+            GUARD_EXIT(s2n_psk_set_hmac(input_psk, input_psk_hmac), "Error setting psk_hmac");
+            GUARD_EXIT(s2n_connection_append_psk(conn, input_psk), "Error appending psk");
+        }
+
         /* See echo.c */
         if (negotiate(conn, sockfd) != 0) {
             /* Error is printed in negotiate */
             S2N_ERROR_PRESERVE_ERRNO();
         }
 
         printf("Connected to %s:%s\n", host, port);
+        if (conn->psk_params.chosen_psk != NULL) {
+            printf("\nPSK Negotiated and a PSK has been choosen.\n");
```

Comment:
Nit: I'm not particularly fond of this phrasing; I would word it like this: "PSK has been negotiated and a PSK has been chosen." Also peep the spelling of chosen.

Reply:
What's the difference between a PSK being negotiated and a PSK being chosen? Aren't those the same thing?